### PR TITLE
Fix Label Generator showing incorrect asset ID

### LINKED ISSUE #289: Label Generator showing incorrect asset ID
### First Check

- [x] This is not a feature request
- [x] I added a very descriptive title to this issue.
- [x] I used the GitHub search to find a similar issue and didn't find it.
- [x] I searched the documentation, with the integrated search.
- [x] I already read the docs and didn't find an answer.
- [x] I can replicate the issue inside the Demo install.

### Homebox Version

v0.15.2

### What is the issue you are experiencing?

I had about 10 assets enters and everything seem to be fine. When I go to the label generator it appears some of my later assets are either not shown in the label generator or have the complete wrong asset id. What might I be doing wrong?

I was able to reproduce on the demo instance. Asset 000-009 and Asset 000-008 are showing incorrect on the label generator page

### How can the maintainer reproduce the issue?

Create new item and go to label generator to see asset doesn't have correct asset id

### Deployment

Docker (Linux)

### OS Architechture

x86_64 (AMD, Intel)

### Deployment Details

_No response_

### PR DESCRIPTION
**Pull request by katosdev**
## What type of PR is this?

- Bug fix

## What this PR does / why we need it:

Fixes Label Generator showing incorrect asset ID due to ordering mismatch. 

## Which issue(s) this PR fixes:

Fixes #289 

## Summary by CodeRabbit

- **New Features**
	- Enhanced user interface for the label generator, improving layout and usability.
	- Repositioned button for generating labels for easier access.

- **Bug Fixes**
	- Improved error handling for page size validation against card dimensions.

- **Documentation**
	- Updated text and formatting for clarity throughout the interface.

### CODE REVIEW

File: frontend/pages/reports/label-generator.vue [hunk 1]
```diff
@@ -1,315 +1,253 @@
...
+  middleware: ["auth"],
+  layout: false,
+});
+useHead({
+  title: "Homebox | Printer",
+});
+
+const api = useUserApi();
+
+const bordered = ref(false);
```
Comment (tonyaellie) [inline frontend/pages/reports/label-generator.vue:14]: This doesn't seem to be used any more.

File: frontend/pages/reports/label-generator.vue [hunk 2]
```diff
@@ -319,135 +257,91 @@
       <h1>Homebox Label Generator</h1>
```
Comment (tonyaellie) [inline frontend/pages/reports/label-generator.vue:257]: Translation keys need to be added for all the text on this page but I think thats should be sorted in a seperate PR?

**Review by tonyaellie — COMMENTED** [review]
image

Seems like its not working properly and the inputs seem to not follow the same styles.

File: frontend/pages/reports/label-generator.vue [hunk 3]
```diff
@@ -1,315 +1,253 @@
...
-        items.push(getItem(i, item as { location: { name: string }; name: string }));
-      } else {
-        items.push(getItem(i, null));
-      }
-    }
-    return items;
+  const items: LabelData[] = [];
+  for (let i = displayProperties.assetRange; i < displayProperties.assetRangeMax; i++) {
+    const item = allFields?.value?.items?.[i];
+    items.push(getItem(i, item?.location ? item : null));
```
Comment (w1ntermute0) [inline frontend/pages/reports/label-generator.vue:178]: This is where the asset ID mismatch happens.  api.items.getAll() which is used to fill allFields appears to return the items in alphabetical order based on name (seems like the entire database, which may be inefficient if you have a large number of items and only want to print a few labels).  Loop index "i" is not associated with the actual assetID of allFields?.value?.items?.[i], it's just whatever order the array happened to get filled in.  This loop index i is what is used to create the QR code and assetID label in "getItem".  Even if the array did happen to get filled in the same order of assetID, this wouldn't handle a case where there was a gap in assetIDs.  

Unfortunately it doesn't look like api.items.getAll() returns an assetID.  Probably the right fix is changing that so it does, or you can look up the assetID based on the id property which is what I did in my comment on issue #289 but that seems more like a band-aid.

**Comment by tankerkiller125** [comment]
Closing in favor of #351

**Pull request closed without merging**